streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -838,4 +838,82 @@ public void punctuate(final long timestamp) {}
         @Override
         public void close() {}
     }
+
+    @Test
+    public void shouldCleanUpPersistentStateStoresOnClose() {
```

Comment:
I think we can simplify this test. We only want to make sure the application directory gets deleted. Thus, we can create it manually, create the driver with corresponding config (passing in a empty Topology), call `close()` and check if the directory was deleted.

Reply:
Yeah, that makes sense, unless you think it might be possible to introduce some other bug later that breaks this behavior via a different mechanism.

Reply:
So, if I read the StateDirectory code right, any subdir of stateDirectory that is parseable as a TaskId and not currently "locked" by a running task gets deleted on `cleanRemovedTasks`.

I could make use of this fact to implement the test as you say, but then this test, which is supposed to be verifying some behavior of the TopologyTestDriver actually becomes coupled to the internal implementation of StateDirectory, which feels wrong to me.

I agree that we don't need a simulation as complete as what I wrote, but it seems like the test should be interacting only with the driver.

Reply:
I'll update the code with what I'm thinking, and you can let me know if you think it's still too silly.